hoodie-client/src/main/java/com/uber/hoodie/io/HoodieAppendHandle.java

```diff
@@ -110,15 +110,15 @@ private void init(HoodieRecord record) {
     if (doInit) {
       this.partitionPath = record.getPartitionPath();
       // extract some information from the first record
-      Optional<FileSlice> fileSlice = fileSystemView.getLatestFileSlices(partitionPath)
-          .filter(fileSlice1 -> fileSlice1.getFileId().equals(fileId)).findFirst();
+      RealtimeView rtView = hoodieTable.getRTFileSystemView();
+      Option<FileSlice> fileSlice = rtView.getLatestFileSlice(partitionPath, fileId);
       // Set the base commit time as the current commitTime for new inserts into log files
       String baseInstantTime = commitTime;
       if (fileSlice.isPresent()) {
         baseInstantTime = fileSlice.get().getBaseInstantTime();
       } else {
         // This means there is no base data file, start appending to a new log file
-        fileSlice = Optional.of(new FileSlice(partitionPath, baseInstantTime, this.fileId));
+        fileSlice = Option.of(new FileSlice(partitionPath, baseInstantTime, this.fileId));
```

Comment:
Option ?

Reply:
is this to overcome Optional not being serializable?